src/shared/Core/Trace2.cs

```diff
@@ -255,6 +271,34 @@ public void Stop(int exitCode, string filePath, int lineNumber)
         });
     }
 
+    public void WriteError(
+        string errorMessage,
+        string formattedErrorMessage = null,
+        [System.Runtime.CompilerServices.CallerFilePath] string filePath = "",
+        [System.Runtime.CompilerServices.CallerLineNumber] int lineNumber = 0)
+    {
+        // It is possible for an error to be thrown before TRACE2 can be initialized.
+        // Since certain dependencies are not available until initialization,
+        // we must immediately return if this method is invoked prior to
+        // initialization.
+        if (!_initialized)
+        {
+            return;
+        }
+
+        WriteMessage(new ErrorMessage()
+        {
+            Event = Trace2Event.Error,
+            Sid = _sid,
+            Time = DateTimeOffset.UtcNow,
+            File = Path.GetFileName(filePath).ToLower(),
```

Comment:
nit: should we be doing ToLower() on the source pathname?   i guess we are in the other methods too.  Feels Windows-centric maybe.   I'm not sure it matters one way or another, TBH.

Reply:
I'm fine with excluding this - I think I originally did this because I thought file names were written as all lowercase per `TRACE2` convention, but now I'm not sure where I got that idea.